kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/db/H2DbServiceImpl.java

```diff
@@ -279,6 +303,12 @@ private void updateInternal(final Map<String, Object> properties) {
                 restartDefragTask(newConfiguration);
             }
 
+            if (this.executorService != null) {
+                this.executorService.shutdown();
```

Comment:
If I understood the documentation correctly, after `shutdown()` is called, the current executor will not accept new tasks and `ExecutorService.submit()` should fail throwing the `RejectedExecutionException` unchecked exception.
We should check that `withConnection()` is not affected by this.